rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/AppVersionRevision.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.util.Version
+import akka.actor.{ ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.cluster.Cluster
+import akka.dispatch.Dispatchers.DefaultBlockingDispatcherId
+import akka.event.Logging
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.{ BootstrapStep, Initializing, NotRunning }
```

Comment:
Should we move the them into a common place?

Reply:
I think those are overkill, should be enough with an `AtomicBoolean`. The only purpose is to only start once, as far as I can see. You can refactor the other places as well.

Reply:
I also removed the `sealed trait BootstrapStep` in ClusterBootstrap, which causes mima to complain. I think it should be ok to add a filter for that. Or Should I bring back this one?

Reply:
that is internal, so adding mima filter is fine